internal/ui/termstatus/terminal_windows.go

```diff
@@ -80,19 +81,47 @@ func isPipe(fd uintptr) bool {
 	return err == nil && typ == windows.FILE_TYPE_PIPE
 }
 
-// canUpdateStatus returns true if status lines can be printed, the process
+func getFileNameByHandle(fd uintptr) (string, error) {
+	type FILE_NAME_INFO struct {
+		FileNameLength int32
+		FileName       [windows.MAX_LONG_PATH]uint16
+	}
+
+	var fi FILE_NAME_INFO
+	err := windows.GetFileInformationByHandleEx(windows.Handle(fd), windows.FileNameInfo, (*byte)(unsafe.Pointer(&fi)), uint32(unsafe.Sizeof(fi)))
+	if err != nil {
+		return "", err
+	}
+
+	filename := syscall.UTF16ToString(fi.FileName[:])
+	return filename, nil
+}
+
+// CanUpdateStatus returns true if status lines can be printed, the process
 // output is not redirected to a file or pipe.
-func canUpdateStatus(fd uintptr) bool {
+func CanUpdateStatus(fd uintptr) bool {
 	// easy case, the terminal is cmd or psh, without redirection
 	if isWindowsTerminal(fd) {
 		return true
 	}
 
-	// check that the output file type is a pipe (0x0003)
+	// pipes require special handling
 	if !isPipe(fd) {
 		return false
 	}
 
-	// assume we're running in mintty/cygwin
-	return true
+	fn, err := getFileNameByHandle(fd)
+	if err != nil {
+		return false
+	}
+
+	// inspired by https://github.com/RyanGlScott/mintty/blob/master/src/System/Console/MinTTY/Win32.hsc
+	// terminal: \msys-dd50a72ab4668b33-pty0-to-master
+	// pipe to cat: \msys-dd50a72ab4668b33-13244-pipe-0x16
+	if (strings.HasPrefix(fn, "\\cygwin-") || strings.HasPrefix(fn, "\\msys-")) &&
+		strings.Contains(fn, "-pty") && strings.HasSuffix(fn, "-master") {
+		return true
+	}
```

Comment:
Maybe add a unit test that asserts that `CanUpdateStatus("\\msys-dd50a72ab4668b33-pty0-to-master")`?

Reply:
I've added a basic test.